File: backend/src/main/java/com/eventflow/modules/refunds/api/dto/RefundDtos.java
package com.eventflow.modules.refunds.api.dto;

import com.eventflow.shared.web.MoneyDto;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.time.Instant;
import java.util.UUID;

/** DTOs espejo EXACTO de los schemas congelados (CreateRefundRequest, RefundResponse, etc.). */
public final class RefundDtos {

    private RefundDtos() {
    }

    public record CreateRefundRequest(@Size(max = 500) String reason) {
    }

    public record RejectRefundRequest(@NotBlank @Size(min = 1, max = 500) String reason) {
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public record RefundResponse(UUID id, UUID ticketId, MoneyDto amount, String status, String reason,
                                 Instant createdAt, Instant resolvedAt) {
    }

    // ===== RecoveryOptionsResponse (tag tickets) =====
    public record RefundQuoteDto(MoneyDto amount, Instant deadline) {
    }

    public record ExchangeQuoteDto(MoneyDto originalPrice, int depreciationPct, MoneyDto listPrice,
                                   Instant listingDeadline) {
    }

    public record RecoveryLinksDto(String action) {
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public record RecoveryOptionsResponse(UUID ticketId, String option, String reason,
                                          RefundQuoteDto refund, ExchangeQuoteDto exchange,
                                          RecoveryLinksDto links) {
    }
}

File: app/src/main/java/com/app/eventflow/data/remote/dto/refunds/RefundDtos.kt
package com.app.eventflow.data.remote.dto.refunds

import com.app.eventflow.data.remote.dto.catalog.CursorMetaDto
import com.app.eventflow.data.remote.dto.catalog.MoneyDto
import kotlinx.serialization.Serializable

/** DTOs espejo de los schemas congelados de Recuperación/Reembolsos (OpenAPI 3.1). */

@Serializable
data class RecoveryOptionsDto(
    val ticketId: String,
    val option: String,                 // REFUND | EXCHANGE | NONE
    val reason: String? = null,
    val refund: RefundQuoteDto? = null,
    val exchange: ExchangeQuoteDto? = null,
    val links: RecoveryLinksDto? = null,
)

@Serializable
data class RefundQuoteDto(val amount: MoneyDto, val deadline: String? = null)

@Serializable
data class ExchangeQuoteDto(
    val originalPrice: MoneyDto,
    val depreciationPct: Int,
    val listPrice: MoneyDto,
    val listingDeadline: String? = null,
)

@Serializable
data class RecoveryLinksDto(val action: String? = null)

@Serializable
data class CreateRefundRequestDto(val reason: String? = null)

@Serializable
data class RejectRefundRequestDto(val reason: String)

@Serializable
data class RefundResponseDto(
    val id: String,
    val ticketId: String,
    val amount: MoneyDto,
    val status: String,                 // REQUESTED | APPROVED | REJECTED | CANCELLED
    val reason: String? = null,
    val createdAt: String,
    val resolvedAt: String? = null,
)

@Serializable
data class RefundsPageDto(val data: List<RefundResponseDto>, val meta: CursorMetaDto)
